bioblend/galaxy/invocations/__init__.py

```diff
@@ -181,6 +213,39 @@ def get_invocation_report(self, invocation_id):
         url = self._make_url(invocation_id) + '/report'
         return self._get(url=url)
 
+    def get_invocation_report_pdf(self, invocation_id, file_path, chunk_size=CHUNK_SIZE):
+        """
+        Get a PDF report for an invocation.
+
+        :type invocation_id: str
+        :param invocation_id: Encoded workflow invocation ID
+
+        :type file_path: str
+        :param file_path: Path to save the file under
+        """
+        url = self._make_url(invocation_id) + '/report.pdf'
+        try:
+            r = self.gi.make_get_request(url, stream=True)
+            r.raise_for_status()
+            with open(file_path, 'wb') as outf:
+                for chunk in r.iter_content(chunk_size):
+                    outf.write(chunk)
+        except Exception as e:
+            raise Exception(f"Failed to get the PDF report, the necessary dependencies may not be installed on the Galaxy server: {e}")
```

Comment:
```suggestion
        r = self.gi.make_get_request(url, stream=True)
        if not r.ok:
            raise Exception(f"Failed to get the PDF report, the necessary dependencies may not be installed on the Galaxy server: {e}")
        with open(file_path, 'wb') as outf:
            for chunk in r.iter_content(chunk_size):
                outf.write(chunk)
           
```

Reply:
I think this is neater. Or is there a reason writing the file needs to be in the try-except block?

Reply:
No, I assumed it was needed because of the get request that could fail. This is definitely neater.